File: tests/benchmark/combinations/ocaml/combinations.ml
(* count combinations of numbers 1 to N having sum N *)

let combinations n =
  let rec go n s =
    if s == 0 then
      1
    else if n == 0 then
      0
    else if s < 0 then
      0
    else
      go n (s - n) + go (n - 1) s
  in
  go n n

;;

print_int (combinations 100);
print_newline ();

File: tests/benchmark/combinations/c/combinations.c
// count combinations of numbers 1 to N having sum N

#include <stdio.h>

int combinations(int n, int s) {
    if (s == 0) {
        return 1;
    } else if (n == 0) {
        return 0;
    } else if (s < 0) {
        return 0;
    } else {
        return combinations(n, s - n) + combinations(n - 1, s);
    }
}

int main() {
    printf("%d\n", combinations(100, 100));
    return 0;
}
